### PR TITLE
bug fix for filter_by None

### PR DESCRIPTION
**Pull request by Jarvi-Izana**
bug fix for https://github.com/apple/turicreate/issues/1716

### CODE REVIEW

**Comment by Jarvi-Izana** [comment]
include irrelevant files. will create a new one.

**Comment by Jarvi-Izana** [comment]
talked with Hoyt. It's okay this time. Next time I will branch from master.

File: src/python/turicreate/data_structures/sframe.py [hunk 1]
```diff
@@ -4408,17 +4408,24 @@ def filter_by(self, values, column_name, exclude=False):
         if column_name not in existing_columns:
             raise KeyError("Column '" + column_name + "' not in SFrame.")
 
+        existing_type = self.column_types()[self.column_names().index(column_name)]
```
Comment (hoytak) [inline src/python/turicreate/data_structures/sframe.py]: Use `self[column_name].dtype`

Comment (Jarvi-Izana) [reply src/python/turicreate/data_structures/sframe.py]: done.

**Pull request merged**